src/accountsdb/db.zig

```diff
@@ -325,7 +326,7 @@ pub const AccountsDB = struct {
         const load_duration = try self.loadFromSnapshot(
             snapshot_fields.accounts_db_fields,
             n_threads,
-            std.heap.page_allocator,
+            allocator,
```

Comment:
we want the page_alloc here because we want a thread-safe and lock-free allocations for fast parallel loading

maybe add a comment to the method parameter mentioning this if were not hard coding it anymore

Reply:
the page_allocator is far from lock-free under contention. memory page pool contention is a large issue that happens when 1. concurrently page allocation and 2. concurrently mmaping. using the provided allocator, that hopefully being the c allocator, will *greatly* increase the speed.

Reply:
nice - it would still be good to include a comment with the given context

Reply:
totally agree with the comment, done